### PR TITLE
code format and update function signatures

### PR DESCRIPTION
**Pull request by Trinkle23897**
Cherry-pick from #200 

- update the function signature
- format code-style
- move _compile into separate functions
- fix a bug in to_torch and to_numpy (Batch)
- remove None in action_range

In short, the code-format only contains function-signature style and `'` -> `"`. (pick up from [black](https://github.com/psf/black))

### CODE REVIEW

File: tianshou/data/utils/converter.py [hunk 1]
```diff
@@ -6,67 +6,71 @@
...
+        return x
     elif isinstance(x, (np.number, np.bool_, Number)):
-        x = np.asanyarray(x)
+        return np.asanyarray(x)
     elif x is None:
-        x = np.array(None, dtype=np.object)
+        return np.array(None, dtype=np.object)
     elif isinstance(x, Batch):
         x.to_numpy()
+        return x
```
Comment (Trinkle23897) [inline tianshou/data/utils/converter.py:26]: I think this inplace operation is a bug.

File: tianshou/data/utils/converter.py [hunk 2]
```diff
@@ -6,67 +6,71 @@
...
+        return x.to(device)
     elif isinstance(x, (np.number, np.bool_, Number)):
-        x = to_torch(np.asanyarray(x), dtype, device)
+        return to_torch(np.asanyarray(x), dtype, device)
     elif isinstance(x, dict):
-        for k, v in x.items():
-            x[k] = to_torch(v, dtype, device)
+        return {k: to_torch(v, dtype, device) for k, v in x.items()}
     elif isinstance(x, Batch):
         x.to_torch(dtype, device)
```
Comment (Trinkle23897) [inline tianshou/data/utils/converter.py:62]: same

File: tianshou/policy/modelfree/ddpg.py [hunk 3]
```diff
@@ -37,20 +37,21 @@ class DDPGPolicy(BasePolicy):
...
+    def __init__(
+        self,
+        actor: Optional[torch.nn.Module],
+        actor_optim: Optional[torch.optim.Optimizer],
+        critic: Optional[torch.nn.Module],
+        critic_optim: Optional[torch.optim.Optimizer],
+        tau: float = 0.005,
+        gamma: float = 0.99,
+        exploration_noise: Optional[BaseNoise] = GaussianNoise(sigma=0.1),
+        action_range: Optional[Tuple[float, float]] = None,
```
Comment (Trinkle23897) [inline tianshou/policy/modelfree/ddpg.py]: should not be None

Comment (duburcqa) [reply tianshou/policy/modelfree/ddpg.py]: Why not ? None could be replaced by [-np.inf, np.inf] automatically. I think it makes sense.

**Review by duburcqa — DISMISSED** [review]
Not much to say about this  PR in practice it changes nothing, but I can see that I don't like much the new formatting style x)

File: tianshou/data/utils/converter.py (hunk 1 above)
Comment (duburcqa) [reply tianshou/data/utils/converter.py:26]: I don't know, it is hard to tell. But in some case, no copy is required and in-place is more efficient. But at least it should be documented.

**Pull request merged**